### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-drawer

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1474]
## Description
This task describes the Storybook implementation for the sd-drawer component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-drawer can be found here: [sd-drawer Notes](https://www.notion.so/matuzo/sd-drawer-88369a4ee9814b9eac0857d8a9636426).

------------------------------------

## Comment
**Problem:** Missing focus management. I assume that sd-drawer doesn't only show automatically but also on user activation. If that happens, you need to decide where focus goes when the user opens the drawer. The drawer is modal, meaning that focus is kinda on the drawer by default, but there are no focus styles.
**Solution:** Manage focus actively. Put focus on the drawer itself or the first focusable element (one of the header buttons).

=> pls also have a look into the "Autofocus" story and make sure it works properly and also add a better name / description like mentioned in the notes below.

**Problem:** If long content doesn't contain interactive elements, it becomes inaccessible to keyboard users in some browsers (e.g. Chromium-based browsers) because scrollable areas are not keyboard accessible by default. (Note: that may change in Chrome 130+, but you'd still have to test it with different browsers).
**Solution:** You could turn the **content area** into a scrollable region using `role=region`, `tabindex=0`, `aria-label`, or `aria-labelledby`.

**Problem:** People who rely on keyboard accessibility have no option to close header-less drawers. Keyboard users can use the Esc key, but switch device users will have difficulty closing.
**Solution:** Close button stays even in no-header variant => absolute positioned like in the sd-dialog

**Problem:** `aria-labelledby="title"` on the panel points to an id that doesn't exist.
**Solution:** removing the attribute and add an a11y hint (in the description of the `no-header` variant) to make users aware of the missing label in case there is no header.

## Notes
- Please note that (screen reader) users will likely miss content coming before an autofocused input field. => add as a11y hint in the story description
- The autofocus example could be better: The accessible name of the input field is “I will have focus when the drawer is opened”.

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
